sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -432,10 +432,26 @@ class Analyzer(override val catalogManager: CatalogManager)
             case e if !e.resolved => u
             case g: Generator => MultiAlias(g, Nil)
             case c @ Cast(ne: NamedExpression, _, _, _) => Alias(c, ne.name)()
-            case e: ExtractValue => Alias(e, toPrettySQL(e))()
+            case e: ExtractValue =>
+              val onlyExtract = e.collect {
+                case _: ExtractValue => true
+                case _ => false
+              }.forall(a => a)
+              if (onlyExtract) {
+                Alias(e, toPrettySQL(e))()
+              } else {
+                val meta = new MetadataBuilder()
+                  .putString("autoGeneratedAlias", "true")
```

Comment:
it's an internal metadata, let's name it `__ autoGeneratedAlias `

Reply:
could we change this to something like `__unstableAlias`? because for example, the alias for `extractOnly` is also auto-generated.